app/ui/browser/views/navbar/location.jsx

```diff
@@ -117,6 +117,15 @@ export class Location extends Component {
     // depending on the eventual UX and if anywhere else would want to change it.
     if (this.props.userTypedLocation !== nextProps.userTypedLocation) {
       this.setState({ focusedResultIndex: -1 });
+
+      // this happens when userTypedLocation changed outside of a 'normal' input
+      // change i.e. 'paste' context menu.  Need to make sure the input is actually set.
+      // No tests for this right now, since shallow rendering doesn't support 'refs'.
```

Comment:
You can use enzyme for non-shallow tests too, which this component should be able to independently test